net/net.go

```diff
@@ -426,29 +416,40 @@ func (n *net) pullThreadUnsafe(ctx context.Context, id thread.ID) error {
 			offsets[lg.ID] = cid.Undef
 		}
 	}
-	var lock sync.Mutex
-	var fetchedRcs []map[peer.ID][]core.Record
-	wg := sync.WaitGroup{}
+
+	var (
+		fetchedRcs []map[peer.ID][]core.Record
+		lock       sync.Mutex
+		wg         sync.WaitGroup
+	)
+
 	for _, lg := range info.Logs {
 		wg.Add(1)
 		go func(lg thread.LogInfo) {
 			defer wg.Done()
 			// Pull from addresses
 			recs, err := n.server.getRecords(ctx, id, lg.ID, offsets, MaxPullLimit)
 			if err != nil {
-				log.Error(err)
+				if errors.Is(err, errConcurrentPull) {
+					log.Debugf("skip updating thread %s (log %s): concurrent pull in progress", id, lg.ID)
+				} else {
+					log.Error(err)
+				}
 				return
 			}
+
 			lock.Lock()
 			fetchedRcs = append(fetchedRcs, recs)
 			lock.Unlock()
 		}(lg)
 	}
 	wg.Wait()
+
+	// maybe we should preliminary deduplicate records?
```

Comment:
As in, check for duplicates in `fetchedRcs`?

Reply:
yes, in order to remove redundant processing of already seen records